tables/endpointsecurity/src/endpointsecurityservice.cpp

```diff
@@ -36,8 +38,13 @@ EndpointSecurityService::~EndpointSecurityService() {
 
   auto status =
       d->virtual_database.unregisterTable(d->process_events_table->name());
-
   assert(status.succeeded() && "Failed to unregister the process_events table");
+
+  if (!d->file_events_table) {
```

Comment:
You need to combine this conditional with the one on l.35; otherwise the function could return early without unregistering the file events table.